File: archive/maximum-product-difference-between-two-pairs.js
const maxProductDifference = function(nums) {
  let maxEl = -Infinity
  let minEl = Infinity

  let prevMax = -Infinity
  let prevMin = Infinity
  nums.forEach(n => {
    if (n >= prevMax) {
      if (n >= maxEl) {
        prevMax = maxEl
        maxEl = n
      } else {
        prevMax = n
      }
    }

    if (n <= prevMin) {
      if (n <= minEl) {
        prevMin = minEl
        minEl = n
      } else {
        prevMin = n
      }
    }
  })

  return Math.abs(prevMin * minEl - maxEl * prevMax)
};

console.log(maxProductDifference([4,2,5,9,7,4,8]))

File: maximum-product-difference-between-two-pairs.go
func maxProductDifference(nums []int) int {
    sort.SliceStable(nums, func (i int, j int) bool {
        return nums[i] < nums[j]
    })

    return nums[len(nums) - 1] * nums[len(nums) - 2] - nums[0] * nums[1]
}